perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java

```diff
@@ -3354,6 +3354,27 @@ public void getGroupMembersCount() throws Exception {
 
 	}
 
+	@Test
+	public void getGroupDirectMembersCount() throws Exception {
+		System.out.println(CLASS_NAME + "getGroupDirectMembersCount");
+
+		vo = setUpVo();
+		Group parentGroup = setUpGroup(vo);
+		Group childGroup = setUpSubgroup(parentGroup);
+
+		Member member = setUpMember(vo);
+
+		groupsManager.addMember(sess, childGroup, member);
+
+		assertThat(groupsManager.getGroupMembersCount(sess, group)).isEqualTo(1);
+
+		assertThat(groupsManager.getGroupDirectMembersCount(sess, group)).isEqualTo(0);
+
+		groupsManager.addMember(sess, parentGroup, member);
+
+		assertThat(groupsManager.getGroupDirectMembersCount(sess, group)).isEqualTo(1);
```

Comment:
It would be clearer if you used the local variable `parentGroup` IMHO. But change it only if you think it's a good idea.
```suggestion
		assertThat(groupsManager.getGroupMembersCount(sess, parentGroup)).isEqualTo(1);

		assertThat(groupsManager.getGroupDirectMembersCount(sess, parentGroup)).isEqualTo(0);

		groupsManager.addMember(sess, parentGroup, member);

		assertThat(groupsManager.getGroupDirectMembersCount(sess, parentGroup)).isEqualTo(1);
```

Reply:
Missed that, thanks